src/nvim/api/ui.c

```diff
@@ -280,6 +280,10 @@ static void remote_ui_mode_change(UI *ui, int mode)
     ADD(args, STRING_OBJ(cstr_to_string("normal")));
   }
   push_call(ui, "mode_change", args);
+
+  Array args2 = ARRAY_DICT_INIT;
+  ADD(args2, INTEGER_OBJ(cursor_shape));
+  push_call(ui, "cursor_shape", args2);
```

Comment:
It is not well-specified what we consider a breaking change. I just know that adding arguments will break some existing clients.